logger/logger.go

```diff
@@ -0,0 +1,23 @@
+package logger
+
+import "log"
+
+// DefaultLogger is the default logger for the client.
+// The client users should provide their own logger implmentations
+type DefaultLogger struct{}
+
+func (l *DefaultLogger) Debugf(format string, args ...interface{}) {
+	log.Printf("[Debug] "+format, args)
```

Comment:
Printf's already string interpolation, so we could pass format and args... to it

Reply:
Is there a way to prepend the log level to the format string without SPrintf/concat? I have updated the implementation with Sprintf for now.

Reply:
use logrus or uber/zap